spec/item_repository_spec.rb

```diff
@@ -341,119 +190,121 @@
         updated_at: Time.now,
         merchant_id: 2
       }
-      new_items = item_repository.create(details2)
-      expect(new_items.last.id).to eq 2
-      expect(item_repository.items.length).to eq 2
+      new_items = item_repository.create(new_item)
+
+      expect(new_items.last.id).to eq 10
+      expect(item_repository.items.length).to eq 11
     end
   end
 
-  describe '#update_id_and_attributes' do
+  describe '#update' do
     it 'updates the item with new attributes' do
-      details1 = {
-        id: 1,
+      details = MockData.items_as_hash
+      mock_data = MockData.items_as_mocks(details) { self }
+      allow_any_instance_of(ItemRepository).to receive(:create_items).and_return(mock_data)
+      item_repository = ItemRepository.new('fake.csv')
+      new_item = {
```

Comment:
⛏ I believe that a cleaner way to do this would be to stub the item.update method -- it would make our test less fragile (since it is currently also relying on the #create method as well as testing the #update method).